src/main/java/com/hubspot/smtp/client/SmtpSession.java

```diff
@@ -71,6 +78,40 @@
     return closeFuture;
   }
 
+  public CompletableFuture<SmtpClientResponse> startTls() {
+    Preconditions.checkState(!isEncrypted(), "This connection is already using TLS");
+
+    return send(new DefaultSmtpRequest(STARTTLS_COMMAND)).thenCompose(r -> {
+      if (SmtpResponses.isError(r)) {
+        return CompletableFuture.completedFuture(r);
+      } else {
+        return performTlsHandshake(r);
+      }
+    });
+  }
+
+  private CompletionStage<SmtpClientResponse> performTlsHandshake(SmtpClientResponse r) {
+    CompletableFuture<SmtpClientResponse> ourFuture = new CompletableFuture<>();
+
+    SslHandler sslHandler = new SslHandler(config.createSSLEngine());
```

Comment:
cool, in addition to being able to provide a `TrustManagerFactory` (I'd add a `@Check` to make sure both weren't present), or instead of that?